elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStore.java

```diff
@@ -8,23 +8,34 @@
 import com.yahoo.elide.core.DataStore;
 import com.yahoo.elide.core.DataStoreTransaction;
 import com.yahoo.elide.core.EntityDictionary;
+import lombok.Getter;
+import lombok.Setter;
 
 /**
  * DataStore that supports Aggregation. Uses {@link QueryEngine} to return results.
  */
 public abstract class AggregationDataStore implements DataStore {
 
+    private QueryEngineFactory queryEngineFactory;
     private QueryEngine queryEngine;
 
-    public AggregationDataStore(QueryEngine queryEngine) {
-        this.queryEngine = queryEngine;
+    @Getter
+    @Setter
+    private EntityDictionary dictionary;
+
+    public AggregationDataStore(QueryEngineFactory queryEngineFactory) {
+        this.queryEngineFactory = queryEngineFactory;
+        queryEngine = null;
     }
 
     @Override
     public abstract void populateEntityDictionary(EntityDictionary dictionary);
 
     @Override
     public DataStoreTransaction beginTransaction() {
+        if (queryEngine == null) {
```

Comment:
Instead of doing this here in the transaction - we should do this in populateEntityDictionary.

Reply:
populateEntityDictionary should:

1. Create the query engine
2. call getSchemas (a new method) on the QueryEngine.
3. Use the schemas returned to bind all the entities.
4. No longer be abstract